test/matrix/test_matrix.rb

```diff
@@ -623,6 +634,120 @@ def test_combine
     assert_equal Matrix.empty(0,3), Matrix.combine(Matrix.empty(0,3), Matrix.empty(0,3)) { raise }
   end
 
+  def test_set_element
+    m1 = Matrix[[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]]
```

Comment:
Tests should be in the same order as the doc.
Successive tests should be independent.
Tests would be more readable if the matrices used were closer to the `[]=` instead of at the beginning of method.